src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,72 @@
+using Flux
+
+basicblock(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 2, pad = 1,
+        BatchNorm(outplanes, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 1, pad = 1,
+        BatchNorm(outplanes, λ=relu))
+
+bottleneck(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 2),
+        BatchNorm(outplanes, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 1),
+        BatchNorm(outplanes, λ = relu))
+
+projection(inplanes::Int, outplanes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+
+identity(inplanes::Int, outplanes::Int, stride::Int) = +
+
+function resnet(channel_config::AbstractArray{Int,1}, block_config::AbstractArray{Int,1})
```

Comment:
Sorry I missed this, can you add `block` and `shortcut` to the input arguments?

Reply:
oh! yes

Reply:
what should be their ``type``? There are 2 different functions for ``block`` so I can't make it a certain type as ``block::bottleneck``. Same case with ``shortcut``

Reply:
You don't need to specify the types. Actually, it probably makes sense to drop the types on the other arguments too.

Reply:
shall i make a function ``Block`` and add both functions i.e ``bottleneck`` and ``basicblock`` in it?

Reply:
ok then!

Reply:
No need. `function resnet(block, shortcut, channel_config, block_config)` is all you need. You don't need to specify types in Julia. It causes no performance changes to leave off the types.

Technically, `basicblock` is a function so it is a subtype of `Function`. But we don't want to specify `::Function` in the argument list, because structs are callable just like functions, but they are not a subtype of `Function`. So, adding the type would restrict people from easily extending the package. In general, it is preferable in Julia not to specify the types unless you have a reason for adding them.